packages/codec/lib/compilations/utils.ts

```diff
@@ -398,51 +398,52 @@ function inferLanguage(
 
 function getIndexToAddAt(
   sourceObject: Source,
-  index: number,
+  index: number | undefined,
   sources: Source[],
   unreliableSourceOrder: boolean
 ): { index: number; needsAdding: boolean; unreliableSourceOrder: boolean } {
-  //first: is this already there? only add it if it's not.
-  //(we determine this by sourcePath if present, and the actual source
-  //contents if not)
   debug("sourcePath: %s", sourceObject.sourcePath);
   debug("given index: %d", index);
   debug(
     "sources: %o",
     sources.map(source => source.sourcePath)
   );
-  if (
-    sources.every(
-      existingSource =>
-        existingSource.sourcePath !== sourceObject.sourcePath ||
+  //first: is this already there? only add it if it's not.
+  //(we determine this by sourcePath if present, and the actual source
+  //contents if not)
+  const existingIndex = sources.findIndex(
+    existingSource =>
+      existingSource &&
+      (existingSource.sourcePath === sourceObject.sourcePath ||
         (!sourceObject.sourcePath &&
           !existingSource.sourcePath &&
-          existingSource.source !== sourceObject.source)
-    )
-  ) {
+          existingSource.source === sourceObject.source))
+  );
+  if (existingIndex === -1) {
+    //it's not already there, let's add it
     if (unreliableSourceOrder || index === undefined || index in sources) {
       //if we can't add it at the correct spot, set the
       //unreliable source order flag
       debug("collision!");
       unreliableSourceOrder = true;
     }
-    //otherwise, just leave things alone
     if (unreliableSourceOrder) {
       //in case of unreliable source order, we'll ignore what indices
       //things are *supposed* to have and just append things to the end
       index = sources.length;
     }
+    //otherwise, just leave things alone
     return {
       index,
       needsAdding: true,
       unreliableSourceOrder
     };
   } else {
-    //return index: null indicates don't add this because it's
-    //already present
     debug("already present, not adding");
+    //note that index may be undefined in this case if it was passed
+    //in as undefined
```

Comment:
Is this true since `index` is now set to `existingIndex` in this clause's return?

Reply:
Oops!  Thanks, this is exactly the sort of thing I meant to fix.  Will fix.